### PR TITLE
Bug 1239185 - Remove use of reference data model in job ingestion

### PR DESCRIPTION
**Pull request by wlach**
WIP, not ready for merging just yet

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1245)

### CODE REVIEW

**Comment by maurodoglio** [comment]
@wlach can you please benchmark the data ingestion process? I'm just worried about the numerous get_or_create calls being a bit slow for big chunks of jobs. Maybe that's not the case, but it would be nice to have numbers confirming that we are good from that point of view.

**Comment by wlach** [comment]
Yeah, will do. I was planning to do a trial run of this on stage before merging to master as well.

**Comment by maurodoglio** [comment]
The description still says it's a wip, but maybe it's ready for a review?

**Comment by wlach** [comment]
It's definitely ready for review, where are you seeing the WIP?

**Comment by maurodoglio** [comment]
https://github.com/mozilla/treeherder/pull/1245#issue-126323407

**Comment by wlach** [comment]
Oh, ok. :) That comment is out of date!

**Comment by maurodoglio** [comment]
cool, I'll have a look today

File: treeherder/model/derived/jobs.py [hunk 1]
```diff
@@ -1148,52 +1155,62 @@ def _load_ref_and_job_data_structs(
...
-
-        job_type_key = self.refdata_model.add_job_type(
-            job_type, job_symbol, group_name, group_symbol
-        )
-
-        product = job.get('product_name', 'unknown')
-        if len(product.strip()) == 0:
-            product = 'unknown'
-        self.refdata_model.add_product(product)
+        (build_platform, _) = BuildPlatform.objects.get_or_create(
```
Comment (maurodoglio) [inline treeherder/model/derived/jobs.py]: there's no need to use parenthesis to unpack variables :dog:

File: treeherder/model/derived/jobs.py [hunk 2]
```diff
@@ -1212,16 +1229,41 @@ def _load_ref_and_job_data_structs(
...
-            reference_data_name, build_system_type, self.project,
-            [build_system_type, self.project, build_os_name, build_platform, build_architecture,
-             machine_os_name, machine_platform, machine_architecture,
-             group_name, group_symbol, job_type, job_symbol,
-             option_collection_hash]
-        )
+        if not reference_data_name:
+            reference_data_name = signature_hash
+
+        (signature, _) = ReferenceDataSignatures.objects.get_or_create(
```
Comment (maurodoglio) [inline treeherder/model/derived/jobs.py]: ReferenceDataSignature has a unique_together index
```
unique_together = ('name', 'signature', 'build_system_type', 'repository')
```
so you should conditionally create the objects based on that, and put all the other fields in the defaults dict

**Comment by maurodoglio** [comment]
I love this patch! :heart_eyes: 

diff_stats

Thanks a lot for working on it. Please let me know how the stage run goes.

**Pull request merged**